lib/async_event_emitter.js

```diff
@@ -99,20 +122,46 @@ module.exports = class AsyncEventEmitter {
    */
   async emit (eventName, ...args) {
     const listeners = this.listeners[eventName]
-    const indexesToRemove = []
 
-    if (!listeners) {
-      return
-    }
+    // TODO: Refactor
+    if (eventName !== CATCH_ALL_EVENT_NAME) {
```

Comment:
Dont have to implement, but it would be cool to have it based on regex. So I could listen for `(*+)_orders`

Reply:
Done!